src/v/kafka/client/test/produce_partition.cc

```diff
@@ -42,11 +43,13 @@ SEASTAR_THREAD_TEST_CASE(test_produce_partition_record_count) {
 
     auto c_res0_fut = producer.produce(make_batch(model::offset(0), 2));
     auto c_res1_fut = producer.produce(make_batch(model::offset(2), 1));
+    ss::maybe_yield().get();
```

Comment:
Originally, after the two produce calls, the consumer would be "in flight" and `handle_response` would be happy.  Now that the consumer enters "in flight" asynchronously, I needed to add in this yield to allow the timer to fire and start the consumer.